### ISSUE TITLE
Upgrade to latest dropwizard metrics

### ISSUE THREAD

**Issue by eranharel** [metrics-clojure/metrics-clojure#133]
The latest dropwizard metrics (4.x.x) removed a lot of clutter, and is now Java 8 based and supports Java 9.
Now that the older (< 8) Java versions are EOL it makes sense to upgrade, and I'll be happy to contribute a patch if it's agreed.

**Comment by michaelklishin** [comment]
Please submit a PR.

**Comment by eranharel** [comment]
I'm already on it @michaelklishin.
I require some input though:
I noticed this project contains some outdated components which were removed from dropwizard metrics, or are no longer maintained. For example the Rieman, influxdb and Ganglia reporters.
If you want to keep support for these components, it will probably require upgrading the target projects. This will require more time than I probably have, as it requires specific knowledge on these components and installations in order to properly test the change.

I guess the decision of whether to keep these components should be based on actual usage, and what you'd like to support.

It may be possible to only make a partial update - still don't know...

**Comment by michaelklishin** [comment]
Are there separate projects for those targets? I find it hard to believe that InfluxDB and Metrics have so few users in common.

I'd like to keep those but if it means maintaining 3 more projects that's probably unrealistic, so we should drop them. Our next release will be a 3.0, so we can do that.

**Comment by eranharel** [comment]
I agree, dropwizard and InfluxDB are both popular projects. InfluxDB is now a part of a commercial product though, which may have reduced some of the users.  
I'm still trying to find an alternative solution for the seemingly dead dependencies.
Will post my finding here next week.

**Comment by eranharel** [comment]
I opened these: 
https://github.com/iZettle/dropwizard-metrics-influxdb/issues/84

**Comment by eranharel** [comment]
I opened these issues: 
https://github.com/iZettle/dropwizard-metrics-influxdb/issues/84
https://github.com/riemann/riemann-java-client/issues/87

Metrics-ganglia is gone. Couldn't find this project, or anything similar.

I'll wait to see how they respond, then decide how to proceed

**Comment by jakemcc** [comment]
@eranharel Looks like those projects did upgrades. Any update on what you were working on?

**Comment by eranharel** [comment]
@jakemcc I started upgrading it on my laptop back then when I opened this ticket. Can't find this work now, but it wasn't that difficult - mainly package renaming and such.

I'll find time to come back to this soon. It shouldn't take me long to do this once I start.

**Comment by otrosien** [comment]
ping.. shall someone take over?

**Comment by michaelklishin** [comment]
@otrosien feel free to submit a PR.

**Comment by atw-gr** [comment]
I think we're close on the InfluxDB and Riemann reporters supporting Java 9+ and using Riemann 4.x:

* https://github.com/iZettle/dropwizard-metrics-influxdb/pull/95 (merged but not yet in a release)
* https://github.com/riemann/riemann-java-client/commit/eaf2e0ee849ff36b470cefc514c8d3f7e3c7c22c (not yet released on Clojars)

metrics-ganglia looks to be dropped for good: https://github.com/dropwizard/dropwizard/pull/2239
